lib/std/os/windows.zig

```diff
@@ -935,6 +936,22 @@ pub fn DeleteFile(sub_path_w: []const u16, options: DeleteFileOptions) DeleteFil
         .CANNOT_DELETE => return error.AccessDenied,
         else => return unexpectedStatus(rc),
     }
+    var file_dispo = FILE_DISPOSITION_INFORMATION{
+        .DeleteFile = TRUE,
+    };
+    rc = ntdll.NtSetInformationFile(
+        tmp_handle,
+        &io,
+        &file_dispo,
+        @sizeOf(FILE_DISPOSITION_INFORMATION),
+        .FileDispositionInformation,
+    );
+    CloseHandle(tmp_handle);
+    switch (rc) {
+        .SUCCESS => return,
+        .DIRECTORY_NOT_EMPTY => return error.DirNotEmpty,
+        else => return unexpectedStatus(rc),
+    }
```

Comment:
Would probably be good to handle all of the errors mentioned in section 4.3.2 (page 33) of https://go.microsoft.com/fwlink/?LinkId=140636:

> 4.3.2 Set Delete-on-closeusing FileDispositionInformationInformation Class (IRP_MJ_SET_INFORMATION)
>
> The FileDispositionInformation information class can be used to mark a link or stream for deletion and transition it to the delete-pending state.  In this state attempts to open the link or stream will fail with STATUS_DELETE_PENDING.
>
> This operation can fail for any of the following reasons:
> - The handle wasopened by ID –STATUS_INVALID_PARAMTER
> - The file is an internal file system metadata file –STATUS_CANNOT_DELETE
> - The file is marked read-only –STATUS_CANNOT_DELETE
> - The volume is marked read-only _-STATUS_MEDIA_WRITE_PROTECTED or STATUS_ACCESS_DENIED (on CDFS & UDF file systems when the file system is read-only)
> - The handle is to a directory that still has files or directories in it –STATUS_DIRECTORY_NOT_EMPTY